src/gausskernel/storage/k2/fdw_handlers.cpp

```diff
@@ -327,9 +328,11 @@ k2BeginForeignScan(ForeignScanState *node, int eflags)
 
 
     /* Allocate and initialize K2PG scan state. */
-    K2FdwExecState *k2pg_state = new K2FdwExecState();
```

Comment:
This won't work I think. K2FdwExecState is a c++ object with nested C++ objects which require a delete call to make sure memory is released recursively.

Reply:
changed as we discussed.